src/App.tsx

```diff
@@ -1,42 +1,131 @@
-import React from 'react';
+/* eslint-disable prettier/prettier */
+import * as dataFromServer from './api/users';
+
+import React, { useEffect, useState } from 'react';
 import 'bulma/bulma.sass';
 import '@fortawesome/fontawesome-free/css/all.css';
-import './App.scss';
 
+import './App.scss';
 import classNames from 'classnames';
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { User } from './types/User';
+import { Post } from './types/Post';
+import { Comment } from './types/Comment';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [currentUser, setCurrentUser] = useState<User | null | undefined>(null);
+  const [currentPost, setCurrentPost] = useState<Post | null | undefined>(null);
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isCommentLoading, setCommentLoading] = useState(false);
+  const [isPostLoaded, setIsPostLoaded] = useState(false);
+  const [isPostLoadError, setIsPostLoadError] = useState(false);
+  const [isCommentLoadError, setIsCommentLoadError] = useState(false);
+  const [isLoadedByAddComment, setIsLoadedByAddComment] = useState(false);
+  const [isOpenForm, setIsOpenForm] = useState(false);
+
+
+  const getCommentsByCurrentPost = (post: Post) => {
+    setIsCommentLoadError(false);
+    setCommentLoading(true);
+    dataFromServer
+      .getCommentsByPost(post?.id)
+      .then(setComments)
+      .catch(()=>setIsCommentLoadError(true))
+      .finally(() => setCommentLoading(false));
+  };
+
+  const getPostsByCurrentUser = (user?: User | null | undefined) => {
+    setIsPostLoadError(false);
+    setIsPostLoaded(true);
+
+    dataFromServer
+      .getPostsByUser(user?.id)
+      .then(setPosts)
+      .catch(() => setIsPostLoadError(true))
+      .finally(() => setIsPostLoaded(false));
+  };
+
+  useEffect(() => {
+    dataFromServer.getUsers().then(setUsers);
+  }, []);
+
+  function addComment(comment: Comment) {
```

Comment:
It is important to stick to one way of creating functions, fix it everywhere
```suggestion
  const addComment = (comment: Comment) => {
```